src/hangar/checkout.py

```diff
@@ -431,7 +443,7 @@ def reset_staging_area(self):
             commit_hash=head_commit)
 
         logger.info(f'Hard reset completed, staging area head commit: {head_commit}')
-        self.__setup()
+        self.close()
```

Comment:
Ok. I'm going to mark it as a TODO for now. There's a bit more logistics that go into handling a reset since we need to invalidate some objects and potentially not others. For now I'm ok with it though.